lib/Dialect/RTL/RTLFolds.cpp

```diff
@@ -706,3 +706,30 @@ void ConcatOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
   };
   results.insert<Folder>(context);
 }
+
+void WireOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
+                                         MLIRContext *context) {
+  struct DropDeadConnect final : public OpRewritePattern<WireOp> {
+    using OpRewritePattern::OpRewritePattern;
+    LogicalResult matchAndRewrite(WireOp op,
+                                  PatternRewriter &rewriter) const override {
+
+      // If this wire is only written to, delete the wire and all writers.
+      SmallVector<Operation *, 4> opsToErase;
+      auto result = op.getResult();
+      for (auto &use : result.getUses()) {
+        Operation *user = use.getOwner();
+        if (!isa<ConnectOp>(user))
+          return failure();
+        opsToErase.push_back(user);
+      }
+
+      for (auto *op : opsToErase)
+        rewriter.eraseOp(op);
+
+      rewriter.eraseOp(op);
+      return success();
+    }
+  };
+  results.insert<DropDeadConnect>(context);
+}
```

Comment:
A couple comments about this code in general:

1. There is probably only ever 1 connect. This would simplify this code even more. @darthscsi [opened a discourse](https://llvm.discourse.group/t/rfc-one-connect-semantics-for-rtl/2572) about this
1. Right now this code removes all connects and the wire.  I could split this in to 2 patterns, removing the writes and then let DCE delete the wire. Any preference?
1. There is probably no reason to build up a vector, we can just iterate through the uses again.  Is there a function to delete all users of a value?
1. We can probably make this pattern generic enough to apply to registers later on.  Are there any other inout types that this would be useful for?

Reply:
erasing an opt should only invalidate use iterators pointing to it, I believe.  Thus you should be able to erase in a second pass without using the vector, if you are careful.

Reply:
I went with this:
```c++
// Remove all uses of the wire.
for (auto &use : op.getResult().getUses())
  rewriter.eraseOp(use.getOwner());
```